gradio/components.py

```diff
@@ -0,0 +1,2412 @@
+        called_directly (bool): if true, the interface was called(), otherwise, it is being used via the GUI
+        """
+        return x
+
+    def preprocess_example(self, x: Any) -> Any:
+        """
+        Any preprocessing needed to be performed on an example before being passed to the main function.
+        """
+        return x
+
+    def set_interpret_parameters(self):
+        """
+        Set any parameters for interpretation.
+        """
+        return self
+
+    def get_interpretation_neighbors(self, x: Any) -> Tuple[List[Any], Dict[Any], bool]:
+        """
+        Generates values similar to input to be used to interpret the significance of the input in the final output.
+        Parameters:
+        x (Any): Input to interface
+        Returns: (neighbor_values, interpret_kwargs, interpret_by_removal)
+        neighbor_values (List[Any]): Neighboring values to input x to compute for interpretation
+        interpret_kwargs (Dict[Any]): Keyword arguments to be passed to get_interpretation_scores
+        interpret_by_removal (bool): If True, returned neighbors are values where the interpreted subsection was removed. If False, returned neighbors are values where the interpreted subsection was modified to a different value.
+        """
+        return [], {}, True
+
+    def get_interpretation_scores(
+        self, x: Any, neighbors: List[Any], scores: List[float], **kwargs
+    ) -> List[Any]:
+        """
+        Arrange the output values from the neighbors into interpretation scores for the interface to render.
+        Parameters:
+        x (Any): Input to interface
+        neighbors (List[Any]): Neighboring values to input x used for interpretation.
+        scores (List[float]): Output value corresponding to each neighbor in neighbors
+        kwargs (Dict[str, Any]): Any additional arguments passed from get_interpretation_neighbors.
+        Returns:
+        (List[Any]): Arrangement of interpretation scores for interfaces to render.
+        """
+        pass
+
+    def generate_sample(self) -> Any:
+        """
+        Returns a sample value of the input that would be accepted by the api. Used for api documentation.
+        """
+        pass
+
+    # Output Functionalities
+    def postprocess(self, y):
+        """
+        Any postprocessing needed to be performed on function output.
+        """
+        return y
+
+    def deserialize(self, x):
+        """
+        Convert from serialized output (e.g. base64 representation) from a call() to the interface to a human-readable version of the output (path of an image, etc.)
+        """
+        return x
+
+
+class Textbox(Component):
+    """
+    Component creates a textbox for user to enter string input or display string output. Provides a string as an argument to the wrapped function.
+    Input type: str
+    Output type: str
+
+    Demos: hello_world, diff_texts, sentence_builder
+    """
+
+    def __init__(
+        self,
+        default: str = "",
```

Comment:
Rename from `default` to `default_value` as per discussion

(this change applies to all components)

Reply:
I am planning to do it after this PR is completed,
But we need to decide on should we have default_value on each component? Because some components feel like not eligible to having a default_value. A few examples:

- json
- highlightedtext
- markdown
- buttown

@aliabid94 @abidlabs

Reply:
`JSON`, `HighlightedText`, and `Markdown` should have a "default value" parameter because that is how we will use them statically. 

E.g.

```
Markdown("""
# My cool demo
Try my **cool demo** below
""")
```

I think the "default value" of a button can just the text that is displayed on it? Or we could have a separate parameter, I think either is fine.

Reply:
Quick comment: what about just dropping "default" and calling the parameter `value`?

Reply:
image value?
video value?
textbox value?

Does not feel intuitive to me, better use default or default_value

Reply:
Ok `default_value` is fine too